src/App.tsx

```diff
@@ -1,22 +1,26 @@
-import React from "react";
-import { ApiPromiseContextProvider } from "context/SubspaceContext";
-import MainLayout from "components/MainLayout";
+import React, { useState } from "react";
 import { WsProvider } from "@polkadot/rpc-provider";
+import { HealthContextProvider, SystemContextProvider } from "context";
+import { ApiPromiseContextProvider } from "context";
+import MainLayout from "layout/MainLayout";
 
-const WS_PROVIDER = process.env.REACT_APP_WS_PROVIDER;
+const WS_PROVIDER = process.env.REACT_APP_WS_PROVIDER || "ws://localhost:9944";
 
 const App = () => {
   if (!WS_PROVIDER) {
-    console.error("WS_PROVIDER not found from env");
-    return null;
+    console.error("WS_PROVIDER not found from env, connecting to a local node");
   }
 
-  const provider = new WsProvider(WS_PROVIDER);
+  const [provider] = useState<WsProvider>(new WsProvider(WS_PROVIDER));
 
   return (
-    <ApiPromiseContextProvider provider={provider}>
-      <MainLayout />
-    </ApiPromiseContextProvider>
+    <SystemContextProvider provider={provider}>
```

Comment:
maybe we can create `ProviderContext` and export `useProvider()` hook wherever we need it

Reply:
I have follow your advice and create a provider context with useProvider hook
Just to comment, want to extract all this contexts pattern to a package we can re use in the future, kind of  

https://github.com/polkadot-js/apps/blob/master/packages/react-hooks
https://github.com/polkadot-js/apps/tree/master/packages/react-api

That way this anidated ContextProvider will dissapear and data be handled on each component wherever we need it using hooks.